lib/scout_apm/store.ex

```diff
@@ -81,20 +81,33 @@ defmodule ScoutApm.Store do
   # Split reporting periods we have into ready & not ready. Ship the ready ones
   # (which stops their process), and next_state has the ones that weren't ready.
   def handle_info(:tick, state) do
-    categorized = Enum.group_by(state.reporting_periods,
+    # Ensure a current reporting period is initialized. Otherwise, samplers won't run unless there is web throughput.
+    {_, new_state} = find_or_create_reporting_period(state)
+    categorized = Enum.group_by(new_state.reporting_periods,
                   fn rp -> StoreReportingPeriod.ready_to_report?(rp) end)
     ready = List.wrap(categorized[:ready])
     not_ready = List.wrap(categorized[:not_ready])
 
     Enum.each(ready, fn rp ->
       Task.start(fn ->
-        StoreReportingPeriod.report!(rp)
+        rp |> capture_samplers |> StoreReportingPeriod.report!
       end)
     end)
 
     schedule_tick()
 
-    {:noreply, %{state | reporting_periods: not_ready}}
+    {:noreply, %{new_state | reporting_periods: not_ready}}
+  end
+
+  # Runs samplers, which should run once per-minute just before reporting.
+  def capture_samplers(reporting_period) do
+    Logger.debug("Capturing samplers")
+    Enum.each([ScoutApm.Instruments.Samplers.Memory], fn sampler ->
+      sampler.metrics |> Enum.each(fn metric ->
+        StoreReportingPeriod.record_metric(reporting_period, metric)
```

Comment:
Also, I'm a little unconvinced that Reporting Period should be an Agent process of its own. It may make sense to make it a normal non-process struct to reduce some of the odd `each` logic (turning it into a `reduce` here)